core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java

```diff
@@ -48,13 +50,16 @@
 import static io.trino.util.HashCollisionsEstimator.estimateNumberOfHashCollisions;
 import static it.unimi.dsi.fastutil.HashCommon.arraySize;
 import static it.unimi.dsi.fastutil.HashCommon.murmurHash3;
+import static java.lang.Math.min;
 import static java.lang.Math.toIntExact;
 import static java.util.Objects.requireNonNull;
 
 // This implementation assumes arrays used in the hash are always a power of 2
 public class MultiChannelGroupByHash
         implements GroupByHash
 {
+    private static final int BATCH_SIZE = 128;
```

Comment:
I will repeat the question from @erichwang (did I mess an answer?). How did you come up with this value? Did you test other?